### ISSUE TITLE
Requesto for new lint: Recommend `uname -n` in place of a plain `hostname`

### ISSUE THREAD

**Issue by slatian** [koalaman/shellcheck#3483]
#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues ~~and didn't find anything related~~ requesting this feature

#### Here's a snippet or screenshot that shows a potential problem:

```sh
#!/bin/sh
printf "You are on host %s" "$(hostname)"
```

#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted to see:

Shellcheck should flag this use of hostname as non POSIX (see also issue #3468) and recommend `uname -n` instead (not that this does not include other usecases of the `hostname` command like setting the hostname).
